rust_snuba/rust_arroyo/src/backends/mod.rs

```diff
@@ -41,7 +41,7 @@ pub enum ProducerError {
 
 /// This is basically an observer pattern to receive the callbacks from
 /// the consumer when partitions are assigned/revoked.
-pub trait AssignmentCallbacks: Send + Sync {
```

Comment:
Why was this Sync ? The assignment callback seems something that should be used on  the main thread only. Which makes me doubt we should need `Send` either.

Reply:
I'm actually not sure why we were able to drop Sync here (i copied this part from https://github.com/getsentry/snuba/pull/4936).

https://docs.rs/rdkafka/latest/rdkafka/client/trait.ClientContext.html seems to suggest that both Sync + Send are required.